code/datums/components/tether.dm

```diff
@@ -1,39 +1,171 @@
+
+	if (!isnull(embed_target))
+		RegisterSignal(embed_target, COMSIG_ITEM_UNEMBEDDED, PROC_REF(on_embedded_removed))
+		RegisterSignal(embed_target, COMSIG_QDELETING, PROC_REF(on_delete))
+
+/datum/component/tether/UnregisterFromParent()
+	UnregisterSignal(parent, COMSIG_MOVABLE_PRE_MOVE)
+	if (!QDELETED(tether_target))
+		UnregisterSignal(tether_target, list(COMSIG_MOVABLE_PRE_MOVE, COMSIG_QDELETING))
+	if (!QDELETED(tether_beam))
+		UnregisterSignal(tether_beam, list(COMSIG_CLICK, COMSIG_QDELETING))
+		qdel(tether_beam)
+	if (!QDELETED(embed_target))
+		UnregisterSignal(embed_target, list(COMSIG_ITEM_UNEMBEDDED, COMSIG_QDELETING))
+
+/datum/component/tether/proc/check_tether(atom/source, new_loc)
 	SIGNAL_HANDLER
 
-	if (get_dist(mover,newloc) > max_dist)
-		to_chat(mover, span_userdanger("The [tether_name] runs out of slack and prevents you from moving!"))
+	if (!isturf(new_loc))
+		to_chat(source, span_warning("[tether_name] prevents you from entering [new_loc]!"))
 		return COMPONENT_MOVABLE_BLOCK_PRE_MOVE
 
+	var/atom/movable/anchor = (source == tether_target ? parent : tether_target)
+	if (get_dist(anchor, new_loc) > cur_dist)
+		if (!istype(anchor) || anchor.anchored || !anchor.Move(get_step_towards(anchor, new_loc)))
+			to_chat(source, span_warning("[tether_name] runs out of slack and prevents you from moving!"))
+			return COMPONENT_MOVABLE_BLOCK_PRE_MOVE
+
 	var/atom/blocker
-	out:
-		for(var/turf/T in get_line(tether_target,newloc))
-			if (T.density)
-				blocker = T
-				break out
-			for(var/a in T)
-				var/atom/A = a
-				if(A.density && A != mover && A != tether_target)
-					blocker = A
-					break out
+	for (var/turf/line_turf in get_line(anchor, new_loc))
+		if (line_turf.density && line_turf != anchor.loc && line_turf != source.loc)
+			blocker = line_turf
+			break
+		for (var/atom/in_turf in line_turf)
+			if (line_turf == anchor.loc || line_turf == source.loc)
+				if ((in_turf.flags_1 & ON_BORDER_1) && (in_turf.dir & get_dir(source, anchor)))
+					blocker = in_turf
+					break
+				continue
+
+			if (in_turf.density && in_turf != source && in_turf != tether_target)
+				blocker = in_turf
+				break
+
 	if (blocker)
-		to_chat(mover, span_userdanger("The [tether_name] catches on [blocker] and prevents you from moving!"))
+		to_chat(source, span_warning("[tether_name] catches on [blocker] and prevents you from moving!"))
 		return COMPONENT_MOVABLE_BLOCK_PRE_MOVE
+
+	/* Kind of jank atm
+	if (get_dist(anchor, new_loc) != cur_dist || !ismovable(source))
+		return
+
+	var/atom/movable/movable_source = source
+	var/datum/drift_handler/handler = movable_source.drift_handler
+	if (isnull(handler))
+		return
+
+	var/angle = get_angle(movable_source, anchor)
+	var/angle_diff = abs(angle - handler.drifting_loop.angle)
+	if (angle_diff > 180)
+		angle_diff = 360 - angle_diff
+
+	var/force = handler.drift_force * sin(angle_diff)
+	movable_source.newtonian_move(angle, TRUE, drift_force = (force * force) / cur_dist)
+	*/
+
```

Comment:
???

Reply:
temporarily commented out until I undraft, then I'll see if I can make it actually work